packages/grid/x-data-grid/src/hooks/features/columnMenu/useGridColumnMenuPreProcessing.ts

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { GridColumnMenuRootProps } from './columnMenuInterfaces';
+import { GridColDef } from '../../../models/colDef/gridColDef';
+import { GridPrivateApiCommunity } from '../../../models/api/gridApiCommunity';
+
+interface GridColumnMenuPreProcessingProps extends GridColumnMenuRootProps {
+  currentColumn: GridColDef;
+}
+
+const useGridColumnMenuPreProcessing = (
```

Comment:
We usually name hooks "preprocessors" if they are registered during state initialization like those here:
https://github.com/mui/mui-x/blob/a6c96b35cf09d9f7132774197fad480cd9bf90fc/packages/grid/x-data-grid-pro/src/DataGridPro/useDataGridProComponent.tsx#L84-L96

So maybe rename?

```suggestion
const useGridColumnMenuComponents = (
```